File: typescript-vitest/src/varar/hello-varar.steps.ts
import { steps } from '@varar/varar'

const { stimulus, sensor } = steps(() => ({ greeting: '', result: 0 }))

stimulus('I greet {string}', (state, name) => ({ ...state, greeting: `Hello, ${name}!` }))

sensor('the greeting should be {string}', (state, _expected) => state.greeting)

stimulus('expression `{int}+{int}`', (state, op1, op2) => ({ ...state, result: op1 + op2 }))

sensor('evaluate to `{int}`', (state, _count) => state.result)

File: kotlin-junit/src/test/kotlin/varar/hello-varar.steps.kt
@file:JvmName("HelloVararSteps")

package varar

import dev.varar.kotlin.sensor
import dev.varar.kotlin.steps
import dev.varar.kotlin.stimulus

data class HelloCtx(val greeting: String = "", val result: Int = 0)

val helloVararSteps =
    steps(::HelloCtx) {
        stimulus("I greet {string}") { name: String -> copy(greeting = "Hello, $name!") }
        sensor("the greeting should be {string}") { _: String -> greeting }
        stimulus("expression `{int}+{int}`") { a: Int, b: Int -> copy(result = a + b) }
        sensor("evaluate to `{int}`") { _: Int -> result }
    }
